lockbox-ios/View/FxAView.swift

```diff
@@ -104,8 +104,10 @@ extension FxAView: WKScriptMessageHandler {
         // Note that this exploit wouldn't be possible if we were using WebChannels; see
         // https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/WebChannel.jsm
         let origin = message.frameInfo.securityOrigin
-        guard origin.`protocol` == self.url?.scheme && origin.host == self.url?.host && origin.port == (self.url?.port ?? 0) else { // swiftlint:disable:this line_length
-            print("Ignoring message - \(origin) does not match expected origin \(self.url?.origin)")
+        guard origin.`protocol` == self.url.scheme,
```

Comment:
I woke up last night screaming that this shouldn't be a direct deref.

You can collect the URL when the webView finishes loading.

There's a method in `WKNavigationDelegate` called 

```swift
func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!)
```